tests/malware_analyzer/pypi/test_wheel_presence.py

```diff
@@ -0,0 +1,185 @@
+                },
+                "downloads": -1,
+                "filename": filename,
+                "has_sig": False,
+                "md5_digest": "9203bbb130f8ddb38269f4861c170d04",
+                "packagetype": "sdist",
+                "python_version": "source",
+                "requires_python": None,
+                "size": 546,
+                "upload_time": "2016-10-13T05:42:27",
+                "upload_time_iso_8601": "2016-10-13T05:42:27.073842Z",
+                "url": f"https://files.pythonhosted.org/packages/de/fa/2fbcebaeeb909511139ce28d \
+                ac4a77ab2452ba72b49a22b12981b2f375b3/{filename}",
+                "yanked": False,
+                "yanked_reason": None,
+            }
+        ]
+    }
+
+    pypi_package_json.get_releases.return_value = release
+    pypi_package_json.get_latest_version.return_value = version
+    pypi_package_json.component.version = None
+    expected_result: tuple[HeuristicResult, dict] = (HeuristicResult.FAIL, {version: [filename]})
+
+    try:
+        actual_result = analyzer.analyze(pypi_package_json)
+    except HeuristicAnalyzerValueError:
+        pytest.fail("Analyzer threw a HeuristicAnalyzerValueError")
+
+    assert actual_result == expected_result
+
+
+def test_analyze_whl_present(pypi_package_json: MagicMock) -> None:
+    """Test for when only .whl is present, so pass"""
+    analyzer = WheelPresenceAnalyzer()
+    version = "0.1.0"
+    filename = "ttttttttest_nester.py-0.1.0.whl"
+
+    release = {
+        version: [
+            {
+                "comment_text": "",
+                "digests": {
+                    "blake2b_256": "defa2fbcebaeeb909511139ce28dac4a77ab2452ba72b49a22b12981b2f375b3",
+                    "md5": "9203bbb130f8ddb38269f4861c170d04",
+                    "sha256": "168bcccbf5106132e90b85659297700194369b8f6b3e5a03769614f0d200e370",
+                },
+                "downloads": -1,
+                "filename": filename,
+                "has_sig": False,
+                "md5_digest": "9203bbb130f8ddb38269f4861c170d04",
+                "packagetype": "bdist_wheel",
+                "python_version": "py2.py3",
+                "requires_python": None,
+                "size": 546,
+                "upload_time": "2016-10-13T05:42:27",
+                "upload_time_iso_8601": "2016-10-13T05:42:27.073842Z",
+                "url": f"https://files.pythonhosted.org/packages/de/fa/2fbcebaeeb909511139ce28d \
+                ac4a77ab2452ba72b49a22b12981b2f375b3/{filename}",
+                "yanked": False,
+                "yanked_reason": None,
+            }
+        ]
+    }
+
+    pypi_package_json.get_releases.return_value = release
+    pypi_package_json.component.version = version
+    expected_result: tuple[HeuristicResult, dict] = (HeuristicResult.PASS, {version: [filename]})
+
+    try:
+        actual_result = analyzer.analyze(pypi_package_json)
```

Comment:
Are you expecting an exception to be raised here? If not, the `try except` should be removed.

Reply:
I'm not expecting an exception, but if the `HeuristicAnalyzerValueError` is thrown the test should fail. I can remove it and just let the error be thrown, which would also indicate a fail too?

Reply:
Yes, you can remove it because if an error is thrown, the test will fail and indicate a bug.